torch/nn/modules/module.py

```diff
@@ -241,6 +223,69 @@ def apply(self, fn):
         fn(self)
         return self
 
+    def apply_tensor(self, fn, to_parameters=True, to_gradients=True,
+                     to_buffers=True, recursive=True):
```

Comment:
Is the generality of the `to_` arguments needed? If you don't need them for your use case, I'd rather drop them for now and add them later if someone shouts.

I'm also troubled as to whether or not `apply_tensor` should be nonrecursive by default. You can always spell the recursive version as `mod.apply(lambda m: m.apply_tensor(fn))`

Reply:
Both good points -- for the second, I would also like users to be explicit by calling apply. I'll make these changes.